### ISSUE TITLE
Required Ruby version mismatch between rubygems.org UI and `gem specification`

### ISSUE THREAD

**Issue by pirj** [rubygems/rubygems#3870]
https://rubygems.org/gems/public_suffix/versions/1.4.0 reports
> Required Ruby Version: None 

So does https://rubygems.org/gems/public_suffix/versions/1.3.3
> Required Ruby Version: None 

However:
```
$ gem specification --remote public_suffix --version 1.4.0
required_ruby_version: !ruby/object:Gem::Requirement
  requirements:
  - - ">="
    - !ruby/object:Gem::Version
      version: 1.9.3

$ gem specification --remote public_suffix --version 1.3.3
required_ruby_version: !ruby/object:Gem::Requirement
  requirements:
  - - ">="
    - !ruby/object:Gem::Version
      version: 1.8.7
```

This issue is related to:

  - [ ] Network problems
  - [x] Installing a library
  - [ ] Publishing a library
  - [ ] The command line `gem`
  - [ ] Other

Here are my current environment details:

```
$ gem env version
3.0.8
```

- [x] I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

**Comment by deivid-rodriguez** [comment]
This could be a data inconsistency between marshalled gemspec and DB, similar to what we fixed in https://github.com/rubygems/rubygems.org/issues/1906?

/cc @sonalkr132

**Comment by sonalkr132** [comment]
> 1.4.0 - February 01, 2014 (59.5 KB)
> 1.3.3 - December 01, 2013 (57.5 KB)

rubygems.org started storing `required_ruby_version` (previously `ruby_version`) from 20 March, 2014 ([source](https://github.com/rubygems/rubygems.org/pull/621#issuecomment-38123374)). These versions were published before this.

I understand now that bundler checks ruby versions during installation this is slightly more important, however, at the time this just seems to be for the UI. ie not as important to go over all gems and backfill the database.
I suppose we can continue with the assumption that the version published after 20 March 2014 supports this. Is this causing any real known issue?

**Comment by pirj** [comment]
I must confess that I often prefer going to https://rubygems.org/gems to check version dependencies.
We have had issues with RSpec CI, but it turns out it was caused by a different issue, https://github.com/rubygems/rubygems/issues/3594. It was confusing to see Bundler spitting error messages that it fails to install something due to an unsatisfied Ruby version constraint while there was no constraint in https://rubygems.org/gems.

When it's [something as verbose as](https://travis-ci.org/github/rspec/rspec-rails/jobs/703303008#L565):
```
Bundler could not find compatible versions for gem "cucumber":
  In Gemfile:
    cucumber (~> 1.3.5)
    aruba (~> 0.14.12) was resolved to 0.14.14, which depends on
      cucumber (>= 1.3.19)
Bundler could not find compatible versions for gem "ffi":
  In Gemfile:
    ffi (~> 1.9.25)
    aruba (~> 0.14.12) was resolved to 0.14.14, which depends on
      ffi (~> 1.9)
Bundler could not find compatible versions for gem "mime-types":
  In Gemfile:
    mime-types (< 3)
    capybara (~> 2.2.0) was resolved to 2.2.1, which depends on
      mime-types (>= 1.16)
Bundler could not find compatible versions for gem "nokogiri":
  In Gemfile:
    nokogiri (= 1.6.8.1)
    capybara (~> 2.2.0) was resolved to 2.2.1, which depends on
      nokogiri (>= 1.3.3)
Bundler could not find compatible versions for gem "rack":
  In Gemfile:
    rack (!= 2.1.0)
    capybara (~> 2.2.0) was resolved to 2.2.1, which depends on
      rack (>= 1.0.0)
    rack-cache (< 1.3.0) was resolved to 1.2, which dependson
      rack (>= 0.4)
    sprockets (~> 3.0) was resolved to 3.7.2, which depends on
      rack (> 1, < 3)
Bundler could not find compatible versions for gem "ruby":
  In Gemfile:
    ruby
    ffi (~> 1.9.25) was resolved to 1.9.25, which depends on
      ruby (>= 1.9)
    nokogiri (= 1.6.8.1) was resolved to 1.6.8.1, which depends on
      ruby (>= 1.9.2)
    sprockets (~> 3.0) was resolved to 3.7.2, which depends on
      rack (> 1, < 3) was resolved to 2.1.4, which depends on
        ruby (>= 2.2.2)
    rspec-mocks was resolved to 3.9.1, which depends on
      ruby (>= 1.8.7)
    sprockets (~> 3.0) was resolved to 3.7.2, which depends on
      ruby (>= 1.9.3)
Bundler could not find compatible versions for gem "sprockets":
  In Gemfile:
    sprockets (~> 3.0)
    rails (~> 3.2.17) was resolved to 3.2.17, which depends on
      actionpack (= 3.2.17) was resolved to 3.2.17, which depends on
        sprockets (~> 2.2.1) 
```
it's quite hard to narrow it down to a single dependency.

No objections on closing this ticket. If the root cause in #3594 is resolved, I guess not having a Ruby constraint to lookup won't be a big deal. I'll add a shorthand alias to `gem specification --remote --version 1.4.0 public_suffix required_ruby_version` and be done with it.

**Comment by deivid-rodriguez** [comment]
In my opinion, if we choose not to fix the old data, we should at least stop displaying the field at all on the website for versions where this information is unreliable.
